src/frontends/ir/src/ir_deserializer.cpp

```diff
@@ -541,9 +542,66 @@ std::shared_ptr<ngraph::Function> XmlDeserializer::parse_function(
         }
     }
 
+    // Read meta data
+    read_meta_data(function, root.child("meta_data"));
+
     return function;
 }
 
+class MetaDataParser : public ov::Meta {
+public:
+    MetaDataParser(const pugi::xml_node& meta) : m_meta_section(meta) {}
+
+    operator const ov::AnyMap&() const override {
+        parse();
+        return m_parsed_data;
+    }
+
+    operator ov::AnyMap&() override {
+        parse();
+        return m_parsed_data;
+    }
+
+private:
+    ov::AnyMap parse_node(const pugi::xml_node& node) const {
+        ov::AnyMap result;
+        auto has_value = [](const pugi::xml_node& node) {
+            auto attr = node.attribute("value");
+            return !attr.empty();
+        };
+        for (const auto& data : node.children()) {
+            if (has_value(data)) {
+                result[data.name()] = XMLParseUtils::GetStrAttr(data, "value");
+            } else {
+                result[data.name()] = parse_node(data);
+            }
+        }
+        return result;
+    }
+
+    void parse() const {
+        if (parsed)
+            return;
```

Comment:
It's kind of tricky. Standard does not impose neither 1 byte size of boolean variable nor all bits zero representation.
I'll it work as expected in this context, but I'd consider using atomic instead.